File: 1-Semestre/Desarrollo de algoritmos/Actividades/Diagramas de flujo-Python (Semana 5)/Ejercicio 4.py
'''
Calculo del área y perímetro de un rectángulo.

Considere las siguientes formulas:
    a) área = base x altura
    b) perímetro = 2 x (base + altura)

Tanto base como altura son ingresados por el usuario.
'''

print("Este programa calcula el área y perímetro de un rectángulo.")
base = int(input("Ingrese la base: "))
altura = int(input("Ingrese la altura: "))
area = 0
perimetro = 0

while base <= 0 or altura <= 0 or base == altura: # Comprueba que la base y altura sean mayores a 0 y diferentes entre sí
    print("[!] La base y la altura deben ser mayores a 0 y diferentes entre sí.")
    base = int(input("Ingrese la base: "))
    altura = int(input("Ingrese la altura: "))

area = base * altura
perimetro = 2 * (base + altura)
print(f"El área es de {area} cm² y el perímetro es de {perimetro} cm.")

File: 2-Semestre/Taller de Programación/Personal/Python-C#/Ejercicio 4.cs
/*
Calculo del área y perímetro de un rectángulo.

Considere las siguientes formulas:
    a) área = base x altura
    b) perímetro = 2 x (base + altura)

Tanto base como altura son ingresados por el usuario.
*/

using System;

namespace Ejercicio_4
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Este programa calcula el área y perímetro de un rectángulo.\n");
            
            double Base;
            double Altura;

            do // Bloque de código para verificar que sea un rectángulo
            {
                do // Bloque de código para preguntar y verificar la base
                {
                    Console.Write("Ingrese la base: ");
                    Base = Convert.ToDouble(Console.ReadLine());

                    if (Base <= 0)
                    {
                        Console.WriteLine("[!] La base no puede ser menor o igual a 0");
                    }
                } while (Base <= 0); // Se repite hasta que la base sea mayor a 0

                do // Bloque de código para preguntar y verificar la altura
                {
                    Console.Write("Ingrese la altura: ");
                    Altura = Convert.ToDouble(Console.ReadLine());

                    if (Altura <= 0)
                    {
                        Console.WriteLine("[!] La altura no puede ser menor o igual a 0");
                    }
                } while (Altura <= 0); // Se repite hasta que la altura sea mayor a 0

                if (Base == Altura)
                {
                    Console.WriteLine("[!] La base y altura no puede ser iguales");
                }
            } while (Base == Altura);

            // Cálculo e impresión del área y perímetro
            double Area = Base * Altura;
            double Perimetro = 2 * (Base + Altura);

            Console.WriteLine($"El área es de {Area} cm²");
            Console.WriteLine($"El perímetro es de {Perimetro} cm");
        }
    }
}